beacon_node/lighthouse_network/gossipsub/src/behaviour.rs

```diff
@@ -1817,6 +1817,15 @@ where
         // Calculate the message id on the transformed data.
         let msg_id = self.config.message_id(&message);
 
+        if let Some(peer) = self.connected_peers.get_mut(propagation_source) {
+            // Record if we received a message that we already sent a IDONTWANT for
+            if peer.dont_send.contains_key(&msg_id) {
```

Comment:
`dont_send` contains the `IDONWANT` messages the peer sent us, so that we can avoid sending them messages they have sent us `IDONTWANT` for. 
Here you are checking if the peer who sent us the message, has also sent previously an `IDONTWANT` message Pawan.

we don't currently keep track of the `IDONTWANT` messages we sent to which peers, you could do it by adapting the ` idontwant_messages_sent_per_topic` you added in this PR to also contain the peers besides the topic. Can you elaborate on why are you adding this Pawan? Is it for scoring purposes?

Nonetheless we should check first if we have metrics enabled, so that we don't spend iterations that are later discarded because the user doesn't have metrics.

Reply:
Ohh yeah did not notice that, thanks! I added an `dont_send_sent` in the `PeerConnections` struct in 
820c4a4.
The reason I want this metric is to see how latency on the mesh peer connections affects the usefulness of IDONTWANTS. 
If our mesh peers send us IDONTWANTS even if we explicitly requested them not to, that could imply implementation bugs or latency on the connections. If latency is the issue, we could employ try sending the idontwants earlier and check if it makes a difference. I think this metric is useful for that feedback.
lmk what you think.